Sources/AddToPlaylistViewController.swift

```diff
@@ -0,0 +1,206 @@
+                let textField = alertController.textFields?.first else { return }
+
+            guard let text = textField.text, text != "" else {
+                DispatchQueue.main.async {
+                    VLCAlertViewController.alertViewManager(title: NSLocalizedString("ERROR_EMPTY_NAME",
+                                                                                     comment: ""),
+                                                            viewController: self)
+                }
+                return
+            }
+            self.delegate?.addToPlaylistViewController(self, newPlaylistWithName: text)
+        }
+        alertController.addAction(cancelButton)
+        alertController.addAction(confirmAction)
+        present(alertController, animated: true, completion: nil)
+    }
+}
+
+// MARK: - Private initializers
+
+private extension AddToPlaylistViewController {
+    private func initViews() {
+        Bundle.main.loadNibNamed("AddToPlaylistView", owner: self, options: nil)
+        view.backgroundColor = PresentationTheme.current.colors.background
+        setupNavigationBar()
+        setupNewPlaylistButton()
+        setupPlaylistCollectionView()
+    }
+
+    private func setupNavigationBar() {
+        navigationItem.leftBarButtonItem = UIBarButtonItem(title: NSLocalizedString("BUTTON_CANCEL",
+                                                                                    comment: ""),
+                                                           style: .done,
+                                                           target: self,
+                                                           action: #selector(dismissView))
+    }
+
+    private func setupNewPlaylistButton() {
+        newPlaylistButton.layer.masksToBounds = true
+        newPlaylistButton.layer.cornerRadius = 10
+        newPlaylistButton.backgroundColor = PresentationTheme.current.colors.orangeUI
+    }
+
+    private func setupPlaylistCollectionView() {
+        playlistCollectionView.delegate = self
+        playlistCollectionView.dataSource = self
+        playlistCollectionView.collectionViewLayout = collectionViewLayout
+        playlistCollectionView.backgroundColor = PresentationTheme.current.colors.background
+
+
+        let cellNib = UINib(nibName: MediaCollectionViewCell.nibName, bundle: nil)
+        playlistCollectionView.register(cellNib,
+                                        forCellWithReuseIdentifier: MediaCollectionViewCell.defaultReuseIdentifier)
+
+    }
+}
+
+// MARK: - UICollectionViewFlowLayout
+
+extension AddToPlaylistViewController: UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView,
+                        layout collectionViewLayout: UICollectionViewLayout,
+                        sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return CGSize(width: collectionView.frame.width - (sidePadding * 2), height: cellHeight)
```

Comment:
does it make sense to move the cellheight into the mediaCollectionViewCell? or is it different for this controller vs mediacategory ?